tests/common.py

```diff
@@ -171,8 +171,11 @@ def mock_service_log(call):  # pylint: disable=unnecessary-lambda
 @ha.callback
 def async_fire_mqtt_message(hass, topic, payload, qos=0):
     """Fire the MQTT message."""
+    if isinstance(payload, str):
+        payload = bytearray(payload, 'utf-8')
```

Comment:
Please use `payload.encode('utf-8')`. That returns a `bytes` object which is immutable.